apps/form-builder/src/components/FormSubmissions/FormSubmissionTable/index.tsx

```diff
@@ -0,0 +1,90 @@
+'use client'; //this component runs with the client and thus requires 'use client'
+import * as React from 'react';
+import { GridValidRowModel } from '@mui/x-data-grid';
+import { DataGrid, GridColDef } from '@mui/x-data-grid';
+import { FormSubmissionResponse } from '@hack4impact-utk/internal-models';
+interface Props {
+  formSubmissions: FormSubmissionResponse[];
+}
+
+//FormSubmissionTable Component that takes in FormSubmissionResponse[]
+export default function FormSubmissionTable(props: Props) {
+  //create the rows for table using the map function that iterates each element in formSubmission.
+  const rows: GridValidRowModel[] = props.formSubmissions
+    .map((submission_element) => {
+      let createdAtDate;
+      let responderEmail;
+      //check if the responderEmail is a string and if it exists. If string doesn't exists, print out Anonymous.
+      if (typeof submission_element.responderEmail === 'string') {
+        if (!submission_element.createdAt) {
+          responderEmail = 'Anonymous';
+        } else {
+          responderEmail = submission_element.responderEmail;
+        }
+      } else {
+        //if not a string, return null
+        return null;
+      }
+
+      if (submission_element.createdAt instanceof Date) {
+        // Check if submission_element.createdAt is a Date object
+        createdAtDate = submission_element.createdAt;
+      } else if (typeof submission_element.createdAt === 'string') {
+        // If not a Date object, then parse the string object to a Date Object
+        const parsedDate = Date.parse(submission_element.createdAt);
+        if (!isNaN(parsedDate)) {
+          //if successfully parsed, then createdAtDate equals the parsedDate.
+          createdAtDate = new Date(parsedDate);
+        } else {
+          //if parsing fails, return null
+          return null;
+        }
+      } else {
+        //if neither a string or Date object, return null
+        return null;
+      }
```

Comment:
This can just be `createdAtDate = new Date(submission_element.createdAt)`. We are guaranteed to have a date string from the database here, so that is why it doesn't need to be checked. Good job for knowing you had to parse the date! :)